### ISSUE TITLE
VRF command batch check witness does not use delegatedStake

### ISSUE THREAD

**Issue by garethtdavies** [MinaProtocol/mina#9134]
While testing the new VRF commands `mina advanced vrf batch-check-witness` it seems to use the value for `totalStake` as `delegatedStake`. As a result, the `thresholdMet` for the command can be true where it should be false.

Output was generated via `mina advanced vrf batch-generate-witness` and the `vrfThreshold` values were appended to this output.

```
mina advanced vrf batch-check-witness
{
  "message": {
    "globalSlot": "42845",
    "epochSeed": "2vaFz7v83bB3zNj4VvKUM3T5uT8mfnoPmZrbuR3hGAEdQyhZriN9",
    "delegatorIndex": 1276
  },
  "publicKey": "B62qpge4uMq4Vv5Rvc8Gw9qSquUYd6xoW1pz7HQkMSHm6h1o7pvLPAN",
  "c": "19416334177459765413057069821807586036057488642590461566665439753889666087258",
  "s": "3295014664252589310792027359262315069687653157993321251267046274461208608725",
  "ScaledMessageHash": [
    "0x115ae01b81465ab5bff171ee701db625839f8aa4faf358242e0bb7b5532d551d",
    "0xfc56879a92ff31ebe3deef9eeab0b13a692ffd6c75c5c5c08593016b43020d13"
  ],
  "vrfThreshold": {
    "delegatedStake": "1732583545194353",
    "totalStake": "825474412840039233"
  }
}
{
  "message": {
    "globalSlot": "42845",
    "epochSeed": "2vaFz7v83bB3zNj4VvKUM3T5uT8mfnoPmZrbuR3hGAEdQyhZriN9",
    "delegatorIndex": 1276
  },
  "publicKey": "B62qpge4uMq4Vv5Rvc8Gw9qSquUYd6xoW1pz7HQkMSHm6h1o7pvLPAN",
  "c":
    "19416334177459765413057069821807586036057488642590461566665439753889666087258",
  "s":
    "3295014664252589310792027359262315069687653157993321251267046274461208608725",
  "ScaledMessageHash": [
    "0x115ae01b81465ab5bff171ee701db625839f8aa4faf358242e0bb7b5532d551d",
    "0xfc56879a92ff31ebe3deef9eeab0b13a692ffd6c75c5c5c08593016b43020d13"
  ],
  "vrfThreshold": {
    "delegatedStake": "18446744073.709551615",
    "totalStake": "18446744073.709551615"
  },
  "vrfOutput": "EiRso5R2XH6Hznddoz2BWDFY1aH4bMPUcqbLVQHWuZFUSWMP47Sk2",
  "vrfOutputFractional": 0.28249260018840877,
  "thresholdMet": true
}
```

I tested this against the output from `mina advanced vrf generate-witness` which does return correctly (and verified this was not a winning slot).

```
mina advanced vrf generate-witness --delegator-index 1276 --epoch-seed 2vaFz7v83bB3zNj4VvKUM3T5uT8mfnoPmZrbuR3hGAEdQyhZriN9 --global-slot 42845 --privkey-path ~/path-to-my-wallet --delegated-stake 1732583545194353 --total-stake 825474412840039233 --generate-outputs true

{
  "message": {
    "globalSlot": "42845",
    "epochSeed": "2vaFz7v83bB3zNj4VvKUM3T5uT8mfnoPmZrbuR3hGAEdQyhZriN9",
    "delegatorIndex": 1276
  },
  "publicKey": "B62qpge4uMq4Vv5Rvc8Gw9qSquUYd6xoW1pz7HQkMSHm6h1o7pvLPAN",
  "c":
    "20675484422149683680235196069821332945427942714551355049709369745710673392919",
  "s":
    "18364293605438900729007207113385842477849131524072774431140826782834002671259",
  "ScaledMessageHash": [
    "0x115ae01b81465ab5bff171ee701db625839f8aa4faf358242e0bb7b5532d551d",
    "0xfc56879a92ff31ebe3deef9eeab0b13a692ffd6c75c5c5c08593016b43020d13"
  ],
  "vrfThreshold": {
    "delegatedStake": "1732583.545194353",
    "totalStake": "825474412.840039233"
  },
  "vrfOutput": "EiRso5R2XH6Hznddoz2BWDFY1aH4bMPUcqbLVQHWuZFUSWMP47Sk2",
  "vrfOutputFractional": 0.28249260018840877,
  "thresholdMet": false
}
```

**Comment by mrmr1993** [comment]
On closer inspection, this seems to be an inconsistency between the JSON `vrfThreshold` (in MINA) and the command line arguments (in NANOMINA). Inserting the appropriate decimal point seems to give the correct result:
```json
{  "message": {    "globalSlot": "42845",    "epochSeed": "2vaFz7v83bB3zNj4VvKUM3T5uT8mfnoPmZrbuR3hGAEdQyhZriN9",    "delegatorIndex": 1276
  },
  "publicKey": "B62qpge4uMq4Vv5Rvc8Gw9qSquUYd6xoW1pz7HQkMSHm6h1o7pvLPAN",
  "c": "19416334177459765413057069821807586036057488642590461566665439753889666087258",
  "s": "3295014664252589310792027359262315069687653157993321251267046274461208608725",
  "ScaledMessageHash": [
    "0x115ae01b81465ab5bff171ee701db625839f8aa4faf358242e0bb7b5532d551d",
    "0xfc56879a92ff31ebe3deef9eeab0b13a692ffd6c75c5c5c08593016b43020d13"
  ],
  "vrfThreshold": {
    "delegatedStake": "1732583.545194353",
    "totalStake": "825474412.840039233"
  }
}
{
  "message": {
    "globalSlot": "42845",
    "epochSeed": "2vaFz7v83bB3zNj4VvKUM3T5uT8mfnoPmZrbuR3hGAEdQyhZriN9",
    "delegatorIndex": 1276
  },
  "publicKey": "B62qpge4uMq4Vv5Rvc8Gw9qSquUYd6xoW1pz7HQkMSHm6h1o7pvLPAN",
  "c":
    "19416334177459765413057069821807586036057488642590461566665439753889666087258",
  "s":
    "3295014664252589310792027359262315069687653157993321251267046274461208608725",
  "ScaledMessageHash": [
    "0x115ae01b81465ab5bff171ee701db625839f8aa4faf358242e0bb7b5532d551d",
    "0xfc56879a92ff31ebe3deef9eeab0b13a692ffd6c75c5c5c08593016b43020d13"
  ],
  "vrfThreshold": {
    "delegatedStake": "1732583.545194353",
    "totalStake": "825474412.840039233"
  },
  "vrfOutput": "EiRso5R2XH6Hznddoz2BWDFY1aH4bMPUcqbLVQHWuZFUSWMP47Sk2",
  "vrfOutputFractional": 0.28249260018840877,
  "thresholdMet": false
}
```

This was hidden by a bug in the [ocaml-integers library](https://github.com/ocamllabs/ocaml-integers), which emits the maximum 64-bit value `18446744073709551615` instead of throwing an error, if the read value doesn't fit in 64 bits.

**Comment by garethtdavies** [comment]
Ah nice, that's easy :) Confirmed this works for me too!

**Comment by mrmr1993** [comment]
Opened a PR against upstream with a fix: ocamllabs/ocaml-integers#32.
